x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -310,6 +323,46 @@ void processResult(AutodetectResult result) {
         }
     }
 
+    void bulkPersistWithRetry(CheckedRunnable<JobResultsPersister.BulkIndexException> bulkRunnable) {
+        int attempts = 0;
+        while(attempts < maximumFailureRetries) {
+            try {
+                bulkRunnable.run();
+                return;
+            } catch (JobResultsPersister.BulkIndexException ex) {
+                if (isDeadOrDying()) {
+                    return;
+                }
+                final int currentAttempt = attempts;
+                LOGGER.trace(
+                    () -> new ParameterizedMessage("[{}] bulk persist failure on attempt [{}] ", jobId, currentAttempt),
+                    ex
+                );
+                attempts++;
+                try {
+                    double backOff = ((1 << attempts) - 1) / 2.0;
+                    Thread.sleep((int)(backOff * 100));
+                } catch (InterruptedException interrupt) {
+                    LOGGER.warn(
```

Comment:
There should be a `Thread.currentThread().interrupt();` in this `catch` block as well as the logging so that the fact this thread was interrupted is not forgotten.